src/main/java/com/exasol/adapter/capabilities/MainCapability.java

```diff
@@ -87,5 +87,57 @@
      *
      * <p>Example: SELECT * FROM t LIMIT 100 OFFSET 10;</p>
      */
-    LIMIT_WITH_OFFSET
+    LIMIT_WITH_OFFSET,
+
+    /**
+     * Support for joins.
+     * If you don't support this capability no joins will be generated in the pushdown.
+     *
+     * <p>Attention: In order to generate joins in the pushdown you additionally
+     *    have to return at least one capability for join type and join contition.</p>
```

Comment:
"condition".

This is not a good idea since you can create illegal combinations. Why not simply drop the `JOIN` capability and check if one of the other two is set. That should give the same result and prevent tight coupling between capabilities.

Reply:
The idea was to have an easy way to disable the join push-down (using EXCLUDED_CAPABILITIES).